relay-general/src/pii/minidumps.rs

```diff
@@ -130,21 +240,64 @@ impl PiiAttachmentsProcessor<'_> {
         data: &mut [u8],
     ) -> Result<bool, ScrubMinidumpError> {
         let file_state = self.state(filename, ValueType::Minidump);
-        let streams = MinidumpData::parse(data)?.streams()?;
+        let items = MinidumpData::parse(data)?.items()?;
         let mut changed = false;
 
-        for StreamDescriptor(stream_type, value_type, range) in streams {
-            let slice = data
-                .get_mut(range)
-                .ok_or(ScrubMinidumpError::InvalidAddress(stream_type))?;
-
+        for item in items {
             // IMPORTANT: Minidump sections are always classified as Pii:Maybe. This avoids to
             // accidentally scrub stack memory with highly generic selectors. TODO: Update the PII
             // system with a better approach.
             let attrs = Cow::Owned(FieldAttrs::new().pii(Pii::Maybe));
 
-            let state = file_state.enter_static("", Some(attrs), Some(value_type));
-            changed |= self.scrub_bytes(slice, &state);
+            match item {
+                MinidumpItem::StackMemory(range) => {
+                    let slice = data
```

Comment:
This stmt appears to be the same between all match arms

Reply:
It's a consequence of me putting the range into the enum variants instead of carrying it outside of it.